client/modules/IDE/reducers/files.js

```diff
@@ -138,33 +144,48 @@ const files = (state, action) => {
       return initialState();
     case ActionTypes.CREATE_FILE: // eslint-disable-line
     {
-      const newState = state.map((file) => {
+      let newState = state.map((file) => {
```

Comment:
The body of this case statement is pretty long, and I wonder if it could be refactored into smaller functions. I know this is also true in other case statements in this file that you didn't work on, but I think this is an opportunity to do a little cleaning up 😄